src/App.tsx

```diff
@@ -1,26 +1,292 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import './styles/todoapp.scss';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+
+import {
+  USER_ID,
+  deleteTodo,
+  getTodos,
+  patchTodo,
+  createTodo,
+} from './api/todos';
+import { todoFilter } from './utils/todoFilter';
+import { FilterType } from './types/FilterType';
+
 import { UserWarning } from './UserWarning';
+import { ErrorNotification } from './components/ErrorNotification';
+import { ErrorType } from './types/ErrorType';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Header } from './components/Header';
+import { TodosList } from './components/TodosList';
+import { Footer } from './components/Footer';
 
 export const App: React.FC = () => {
+  const [currentTodos, setCurrentTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [deletingTodoId, setDeletingTodoId] = useState<number | null>(null);
+  const [filter, setFilter] = useState(FilterType.All);
+
+  const [errorMessage, setErrorMessage] = useState('');
+  const [showError, setShowError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [idsProccesing, setIdsProccesing] = useState<number[]>([]);
+
+  const [title, setTitle] = useState('');
+
+  const focusField = useRef<HTMLInputElement>(null);
+
+  const beforeRequest = () => {
+    setShowError(false);
+    setIsLoading(true);
+  };
+
+  const handleError = (message: string) => {
+    setErrorMessage(message);
+    setShowError(true);
```

Comment:
i think you can remove this. Because you can use errorMessage, if errorMessage doesn't equal null show error.

Reply:
I agree! Such solution seems more logical. And if I implement it in this way, it works well visually, but for some reason it doesn't pass the tests. I tried to find the reason for this, but unfortunately, without success. So for now, I left the setShowError state just to be able to make all the other adjustments.